Sources/Utils/GeneratableObjectCache.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  GeneratableObjectCache.swift
+//  Alicerce
+//
+//  Created by Filipe Lemos on 02/05/2018.
+//  Copyright © 2018 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public protocol GeneratableObjectKey: Hashable {
```

Comment:
Not sure about this naming – `GeneratableObjectKey` and `GeneratableObjectCache`. `Generatable` is not a very common name and in the Swift ecosystem the common term used is `Generator`.

What do you think @filipe-lemos  and @p4checo?

Reply:
Personally I dislike the term `Generator` used in this context. While Swift doesn't use the term generator related to iterators, I still think those are deeply tied in general.